ios/FluentUI/Navigation/UIBarButtonItem+BadgeValue.swift

```diff
@@ -24,13 +25,21 @@ import UIKit
         }
     }
 
-    /// Convenience method to set the badge value to a number.
-    /// If the number is zero, the badge value will be hidden.
-    @objc func setBadgeNumber(_ number: UInt) {
-        if number > 0 {
-            badgeValue = NumberFormatter.localizedString(from: NSNumber(value: number), number: .none)
-        } else {
-            badgeValue = nil
+    @objc internal var badgeAccessibilityLabel: String? {
+        get {
+            return objc_getAssociatedObject(self, &AssociatedKeys.badgeAccessibilityLabel) as? String ?? nil
+        }
+        set {
+            objc_setAssociatedObject(self, &AssociatedKeys.badgeAccessibilityLabel, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
         }
     }
+
+    /// Use this method on bar button item's instance to set the badge value and badge accessibility label.
+    /// Set the badge value to nil to hide the red oval.
+    /// Complete badge accessibility label would be the concatenation of the badge value and badge accessibility label.
+    /// Complete accessibility label for the item in the navigation bar would be item's accessibility label concatenated with the complete badge accessibility label.
+    @objc func setBadgeValue(_ badgeValue: String?, badgeAccessibilityLabel: String?) {
+        self.badgeAccessibilityLabel = (badgeValue ?? "").appending(badgeAccessibilityLabel ?? "")
```

Comment:
why is this logic duplicated here and in line 218?
It seems to me that this API should only set the properties with the parameter values.

Reply:
Actually, my design was that we are going to ask the user only the suffix string and we will put badge value on our own but now I realized that was not good instead we should be asking for the complete badge accessibility label from the user. So yes, now I have fixed it.